controllers/repository_controller.go

```diff
@@ -143,6 +145,24 @@ func (r *RepositoryReconciler) checkInitial(ctx context.Context, logger logr.Log
 
 	if v, ok := instanceDeepCopy.Labels[corev1alpha1.RepositoryTypeLabel]; !ok || v != instanceDeepCopy.Spec.RepositoryType {
 		instanceDeepCopy.Labels[corev1alpha1.RepositoryTypeLabel] = instanceDeepCopy.Spec.RepositoryType
+		if !instanceDeepCopy.IsOCI() {
+			url := instanceDeepCopy.Spec.URL
+			var auth string
```

Comment:
@zqq454224016  It would be much better if you move all these code inner `getResponse` which I think you should rename to `getRepoType` or else.